### ISSUE TITLE
match_layers.py: "Rebuild state dict failed"

### ISSUE THREAD

**Issue by eschmidbauer** [luigisaetta/whisper-app#2]
Hi-
I followed along your thread from [whisper.cpp](https://github.com/ggerganov/whisper.cpp/issues/157)
I'm trying to convert a fine-tuned huggingface whisper model for use with whisper.cpp
[Specifically this model ](https://huggingface.co/sanchit-gandhi/whisper-medium-switchboard-5k/tree/main)
I've downloaded the [model file](https://huggingface.co/sanchit-gandhi/whisper-medium-switchboard-5k/blob/main/medium.en.whisper) into `./whisper-app/model-custom.pt`
then i run:
```bash
python3 match_layers.py

Loading vanilla Whisper model
Loading vanilla HF Model
Matching layers...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 947/947 [00:17<00:00, 54.11it/s]
Number of keys: 947
Number of sanity_check ok: 947

Match is complete !!!

Serializing map_dict...
map_dict saved as: map_dict.pkl...

Test if it works...

Reloading map_dict...

Loading fine tuned dict...
Rebuild the state dict...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 947/947 [00:00<00:00, 1289193.73it/s]
Traceback (most recent call last):
  File "/root/whisper-app/match_layers.py", line 189, in <module>
    assert n_except == 0, "Rebuild state dict failed"
AssertionError: Rebuild state dict failed```
```

As you can see it fails at rebuilding the state dict.
I would appreciate any help on this, and thank you for sharing your work!

**Comment by luigisaetta** [comment]
Hi @eschmidbauer there are some problems with that model file
1. It has not been trained using DDP: this is not really a problem, we need only in my utility to set DDP_TRAINED = False

But then we see the bigger problem: it seems that this model has layers with different names with the ones that you get fine tuning a model following https://huggingface.co/blog/fine-tune-whisper
And, worse, I'm not able to load that model using

```
hugging_face_model = WhisperForConditionalGeneration.from_pretrained(
    model_name_from_hf)
```
It says that config.json is missing (and in fact it is)

In conclusion, that model files is really out of the scope I have set for my utility and I don't understand what is really is.

One question: why you want to work with that one? 
Maybe, You could ask to the author why the names of the layers are in some way different and if we can workaround it.
Sanchiit is well aware of these cross-loading efforts... he has answered to one of my post... maybe if he has time.

**Comment by eschmidbauer** [comment]
> One question: why you want to work with that one? Maybe, You could ask to the author why the names of the layers are in some way different and if we can workaround it. Sanchiit is well aware of these cross-loading efforts... he has answered to one of my post... maybe if he has time.

@luigisaetta We would like to test inference with `whisper.cpp` using a fine-tuned model on the switchboard dataset. It seems the ones available on huggingface may not work though. I will look into fine-tuning our own model. Thank you for looking into this!

**Comment by luigisaetta** [comment]
@eschmidbauer switchboard dataset, now I understand. What seems to me strange is that Sanchiit published the article describing how to do fine tuning, and for my custom-models I followed his article. Probably this. (switchboard) model was made by him before.

**Comment by eschmidbauer** [comment]
@luigisaetta Im working on fine-tuning using the script in Sanchiit's [repo](https://huggingface.co/sanchit-gandhi/whisper-medium-switchboard-5k/blob/main/run_switchboard.sh)

Once I have a fine-tuned model - i'll give your script a try again, thanks again for sharing your work!
